pagerduty/resource_pagerduty_user_notification_rule_test.go

```diff
@@ -0,0 +1,132 @@
+		CheckDestroy: testAccCheckPagerDutyUserNotificationRuleDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckPagerDutyUserNotificationRuleContactMethodConfig(contactMethodType1),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckPagerDutyUserNotificationRuleExists("pagerduty_user_notification_rule.foo"),
+				),
+			},
+			{
+				Config: testAccCheckPagerDutyUserNotificationRuleContactMethodConfig(contactMethodType2),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckPagerDutyUserNotificationRuleExists("pagerduty_user_notification_rule.foo"),
+				),
+			},
+			{
+				Config: testAccCheckPagerDutyUserNotificationRuleContactMethodConfig(contactMethodType3),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckPagerDutyUserNotificationRuleExists("pagerduty_user_notification_rule.foo"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckPagerDutyUserNotificationRuleDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*pagerduty.Client)
+	for _, r := range s.RootModule().Resources {
+		if r.Type != "pagerduty_user_notification_rule" {
+			continue
+		}
+
+		if _, _, err := client.Users.GetNotificationRule(r.Primary.Attributes["user_id"], r.Primary.ID); err == nil {
+			return fmt.Errorf("User notification rule still exists")
+		}
+
+	}
+	return nil
+}
+
+func testAccCheckPagerDutyUserNotificationRuleExists(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No user notification rule ID is set")
+		}
+
+		client := testAccProvider.Meta().(*pagerduty.Client)
+
+		found, _, err := client.Users.GetNotificationRule(rs.Primary.Attributes["user_id"], rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		if found.ID != rs.Primary.ID {
+			return fmt.Errorf("Notification rule not found: %v - %v", rs.Primary.ID, found)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckPagerDutyUserNotificationRuleContactMethodConfig(methodType string) string {
+	return fmt.Sprintf(`
+resource "pagerduty_user_notification_rule" "foo" {
+  user_id                = "${pagerduty_user.foo.id}"
+  start_delay_in_minutes = 1
+  urgency                = "high"
+
+  contact_method {
+    type = "%[1]v"
+    id   = "${pagerduty_user_contact_method.%[1]v.id}"
+  }
+}
+
+resource "pagerduty_user" "foo" {
+  name        = "foo bar"
+  email       = "foo@bar.com"
+  color       = "red"
+  role        = "user"
+  job_title   = "bar"
+  description = "bar"
+}
+
+resource "pagerduty_user_contact_method" "email_contact_method" {
+  user_id = "${pagerduty_user.foo.id}"
+  type    = "email_contact_method"
+  address = "foo-1@bar.com"
+  label   = "Work"
+}
+
+resource "pagerduty_user_contact_method" "sms_contact_method" {
+  user_id      = "${pagerduty_user.foo.id}"
+  type         = "sms_contact_method"
+  address      = "1234567890"
```

Comment:
When running acceptance tests, this is currently rejected by the Pagerduty API:

```
make testacc TEST=./pagerduty TESTARGS='-run=TestAccPagerDutyUserNotification -count=1'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./pagerduty -v -run=TestAccPagerDutyUserNotification -count=1 -timeout 120m
=== RUN   TestAccPagerDutyUserNotificationRule_import
--- FAIL: TestAccPagerDutyUserNotificationRule_import (7.12s)
    testing.go:538: Step 0 error: Error applying: 2 errors occurred:
                * pagerduty_user_contact_method.phone_contact_method: 1 error occurred:
                * pagerduty_user_contact_method.phone_contact_method: POST API call to https://api.pagerduty.com/users/PFKZX66/contact_methods failed 400 Bad Request. Code: 2001, Errors: [Phone number does not seem to be valid Please contact support for assistance], Message: Invalid Input Provided


                * pagerduty_user_contact_method.sms_contact_method: 1 error occurred:
                * pagerduty_user_contact_method.sms_contact_method: POST API call to https://api.pagerduty.com/users/PFKZX66/contact_methods failed 400 Bad Request. Code: 2001, Errors: [Phone number does not seem to be valid Please contact support for assistance], Message: Invalid Input Provided




=== RUN   TestAccPagerDutyUserNotificationRuleContactMethod_Basic
--- FAIL: TestAccPagerDutyUserNotificationRuleContactMethod_Basic (6.78s)
    testing.go:538: Step 0 error: Error applying: 2 errors occurred:
                * pagerduty_user_contact_method.sms_contact_method: 1 error occurred:
                * pagerduty_user_contact_method.sms_contact_method: POST API call to https://api.pagerduty.com/users/PCOOWTG/contact_methods failed 400 Bad Request. Code: 2001, Errors: [Phone number does not seem to be valid Please contact support for assistance], Message: Invalid Input Provided


                * pagerduty_user_contact_method.phone_contact_method: 1 error occurred:
                * pagerduty_user_contact_method.phone_contact_method: POST API call to https://api.pagerduty.com/users/PCOOWTG/contact_methods failed 400 Bad Request. Code: 2001, Errors: [Phone number does not seem to be valid Please contact support for assistance], Message: Invalid Input Provided




FAIL
FAIL    github.com/terraform-providers/terraform-provider-pagerduty/pagerduty   13.911s
FAIL
make: *** [GNUmakefile:17: testacc] Error 1
```

Using `8005551234` works:

```
make testacc TEST=./pagerduty TESTARGS='-run=TestAccPagerDutyUserNotification -count=1'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./pagerduty -v -run=TestAccPagerDutyUserNotification -count=1 -timeout 120m
=== RUN   TestAccPagerDutyUserNotificationRule_import
--- PASS: TestAccPagerDutyUserNotificationRule_import (11.57s)
=== RUN   TestAccPagerDutyUserNotificationRuleContactMethod_Basic
--- PASS: TestAccPagerDutyUserNotificationRuleContactMethod_Basic (19.69s)
PASS
ok      github.com/terraform-providers/terraform-provider-pagerduty/pagerduty   31.272s
```


```suggestion
  address      = "8005551234"
```